### ISSUE TITLE
OptimFROG blind tags

### ISSUE THREAD

**Issue by Agagamand** [Zeugma440/atldotnet#249]
## The problem

I copied the tags (CopyMetadataTo) from .tak to .ofr, but MP3Tag and foobar2000 don't see the tags in .ofr. Only Aimp does, but only partially.

I want that after copying tags to .ofr, all programs that support APEv2 tags can read them.

## Environment

* ATL version (or git revision) that exhibits the issue: 5.18.0
* OS/version used to run ATL: Win10

**Comment by Zeugma440** [comment]
Alright, thanks for your feedback. Talking about optimFROG is almost necromancy xD I'll see what I can do

**Comment by Agagamand** [comment]
OptimFROG still provides better compression. As long as it remains so, it cannot be considered obsolete.

The necromancy is to use something like "RKau lossless compressor", which is not the best at anything. But that didn't stop the ffmpeg team from adding its support in version 6.0.

**Comment by Zeugma440** [comment]
Looks like there are multiple factors at play here :

- When writing to an audio file without any metadata, ATL uses the `Settings.DefaultTagsWhenNoMetadata` property to determine which tag types it should write. Default value is `{ MetaDataIOFactory.TagType.ID3V2, MetaDataIOFactory.TagType.NATIVE}`, which means **ATL writes ID3v2 values to empty OFR files**.

NB : "Native" means "any non-standard tagging system native to the audio format" such as what MP4/M4A and WMA use.

- MP3tag seems to be unable to read ID3v2 data from OFR files. The [latest foobar2000 OFR plugin](https://github.com/marc2k3/foo_input_ofr/releases/tag/1.43) does, though.

- When changing the default settings to write APE tags on empty files, everything seems to work fine
```cs
Settings.DefaultTagsWhenNoMetadata = new[] { MetaDataIOFactory.TagType.APE };
```

Can you confirm that on your side?

**Comment by Agagamand** [comment]
I confirm.

**Comment by Zeugma440** [comment]
Great suggestion - I've just done that 😄 

Now the default for `Settings.DefaultTagsWhenNoMetadata` is `MetaDataIOFactory.TagType.RECOMMENDED`, which adapts to every audio format.

**Comment by Zeugma440** [comment]
Available in today's v5.19

Please close the issue if it works on your side~
